pkg/apis/sparkoperator.k8s.io/v1beta1/types.go

```diff
@@ -478,8 +478,12 @@ type PrometheusSpec struct {
 	// Optional.
 	// If not specified, 8090 will be used as the default.
 	Port *int32 `json:"port"`
+	// ConfigFile is the path to the custom Prometheus configuration file provided in the Spark job image.
+	// ConfigFile takes precedence over Configuration, which is shown below.
+	ConfigFile string `json:"configFile"`
```

Comment:
This should have type `*string` to signal it's optional. We also need to add tag `omitempty` for this and the field below.